x-pack/legacy/plugins/spaces/public/lib/copy_saved_objects_to_space/summarize_copy_result.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ProcessedImportResponse, SavedObjectRecord } from 'ui/management/saved_objects_management';
+
+export interface SummarizedSavedObjectResult {
+  type: string;
+  id: string;
+  name: string;
+  conflicts: ProcessedImportResponse['failedImports'];
+  missingReferences: ProcessedImportResponse['failedImports'];
+  hasUnresolvableErrors: boolean;
+}
+
+interface SuccessfulResponse {
+  successful: true;
+  hasConflicts: false;
+  hasUnresolvableErrors: false;
+  objects: SummarizedSavedObjectResult[];
+  processing: false;
+}
+interface UnsuccessfulResponse {
+  successful: false;
+  hasConflicts: boolean;
+  hasUnresolvableErrors: boolean;
+  objects: SummarizedSavedObjectResult[];
+  processing: false;
+}
+
+interface ProcessingResponse {
+  objects: SummarizedSavedObjectResult[];
+  processing: true;
+}
+
+export type SummarizedCopyToSpaceResult =
+  | SuccessfulResponse
+  | UnsuccessfulResponse
+  | ProcessingResponse;
+
+export function summarizeCopyResult(
```

Comment:
If we only use this from within `x-pack/legacy/plugins/spaces/public/views/management/components/copy_saved_objects_to_space` should we move it there?

Reply:
We can...I chose to put it here so that I could test it independently from the React components, and it felt wrong to have this non-component file within the `**/components/**` directory structure